media-api/app/lib/elasticsearch/sorts.scala

```diff
@@ -1,17 +1,44 @@
 package lib.elasticsearch
 
+import lib.querysyntax._
 import org.elasticsearch.action.search.SearchRequestBuilder
-import org.elasticsearch.search.sort.SortOrder
+import org.elasticsearch.index.query.{FilterBuilders}
+import org.elasticsearch.search.sort.{SortBuilders, SortOrder}
 
 
 object sorts {
 
-  def parseFromRequest(sortBy: Option[String])(builder: SearchRequestBuilder): SearchRequestBuilder = {
+  def createSort(sortBy: Option[String], query: List[Condition])(builder: SearchRequestBuilder) = {
+    sortBy match {
+      case Some("dateAddedToCollection") => addedToCollectionTimeSort(query)(builder)
+      //for any other sortBy term
+      case _ => uploadTimeSort(sortBy)(builder)
+    }
+  }
+
+  def uploadTimeSort(sortBy: Option[String])(builder: SearchRequestBuilder): SearchRequestBuilder = {
     val sorts = sortBy.fold(Seq("uploadTime" -> SortOrder.DESC))(parseSorts)
     for ((field, order) <- sorts) builder.addSort(field, order)
     builder
   }
 
+  def addedToCollectionTimeSort(query: List[Condition])(builder: SearchRequestBuilder): SearchRequestBuilder = {
+    //TODO get relevant values from query
+    val pathHierarchyOpt = query.map {
+      case Match(HierarchyField, Phrase(value)) => Some(value)
+      case _ => None
+    }.flatten.headOption
```

Comment:
If something is a `Match(HierarchyField, Phrase(value))` then it has to be a collection query. This is because this type is only constructed once we have identified it as a collection query. So we use it to pattern match and return the pathId it contains.